haystack/preview/components/generators/hugging_face/hugging_face_local.py

```diff
@@ -0,0 +1,136 @@
+    ):
+        """
+        :param model_name_or_path: The name or path of a Hugging Face model for text generation.
+            E.g. "google/flan-t5-large".
+            If the model is also specified in the pipeline_kwargs, this parameter will be ignored.
+        :param task: The task for the Hugging Face pipeline.
+            Possible values are "text-generation" and "text2text-generation".
+            Generally, decoder-only models like GPT support "text-generation",
+            while encoder-decoder models like T5 support "text2text-generation".
+            If the task is also specified in the pipeline_kwargs, this parameter will be ignored.
+            If not specified, the component will attempt to infer the task from the model name,
+            calling the Hugging Face Hub API.
+        :param device: The device on which the model is loaded. (e.g., "cpu", "cuda:0").
+            If device or device_map is specified in the pipeline_kwargs, this parameter will be ignored.
+        :param token: The token to use as HTTP bearer authorization for remote files.
+            If True, will use the token generated when running huggingface-cli login (stored in ~/.huggingface).
+            If the token is also specified in the pipeline_kwargs, this parameter will be ignored.
+        :param generation_kwargs: a dictionary containing keyword arguments to customize text generation.
+            See Hugging Face's documentation for more information.
+            https://huggingface.co/docs/transformers/main/en/generation_strategies#customize-text-generation
+            https://huggingface.co/docs/transformers/main/en/main_classes/text_generation#transformers.GenerationConfig
+        :param pipeline_kwargs: Dictionary containing keyword arguments used to initialize the pipeline.
+            These keyword arguments provide fine-grained control over the pipeline.
+            In case of duplication, these kwargs override model_name_or_path, task, device, and token init parameters.
+            See Hugging Face's documentation for more information on the available kwargs.
+            https://huggingface.co/docs/transformers/en/main_classes/pipelines#transformers.pipeline.task
+            In this dictionary, you can also include "model_kwargs" to specify the kwargs
+            for model initialization:
+            https://huggingface.co/docs/transformers/en/main_classes/model#transformers.PreTrainedModel.from_pretrained
+        """
+        transformers_import.check()
+
+        pipeline_kwargs = pipeline_kwargs or {}
+        generation_kwargs = generation_kwargs or {}
+
+        # check if the pipeline_kwargs contain the essential parameters
+        # otherwise, populate them with values from other init parameters
+        pipeline_kwargs.setdefault("model", model_name_or_path)
+        pipeline_kwargs.setdefault("token", token)
+        if device is not None and "device" not in pipeline_kwargs and "device_map" not in pipeline_kwargs:
```

Comment:
What happens if device is None and it is nowhere specified, will it default to "cpu" and work as well?

Reply:
Yes. (I tried).